controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -589,7 +589,7 @@ func (w *Workload) UpdateKubeProxyImageInfo(ctx context.Context, kcp *controlpla
 	if len(ds.Spec.Template.Spec.Containers) == 0 {
 		return nil
 	}
-	newImageName, err := util.ModifyImageTag(ds.Spec.Template.Spec.Containers[0].Image, kcp.Spec.Version)
+	newImageName, err := util.ModifyImageTag(ds.Spec.Template.Spec.Containers[0].Image, util.NormalizedKubernetesBuildVersion(kcp.Spec.Version))
```

Comment:
While we're at it, can we find the actual container like we do in https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/internal/workload_cluster_coredns.go#L276-L283

Reply:
Yup. Totally reasonable.